quote-book/tests/common/mod.rs

```diff
@@ -298,3 +327,39 @@ pub fn get_quotes_filtering_works(quote_book: &impl QuoteBook) {
         ]
     );
 }
+/// Adds a block containing the given keyimage to the ledger and returns the new
+/// block.
+///
+/// # Arguments
+/// # TODO(wjuan): This should be refactored in along with the test_utils for adding to ledger in the mobilecoin repo.
+/// * `ledger` - Ledger instance.
+/// * `block_version` - The block version to use.
+/// * `key image` - The key-image to be added
+/// * `rng` - Random number generator.
+pub fn add_key_image_to_ledger(
```

Comment:
Is there a reason not to use the stuff inside https://github.com/mobilecoinfoundation/mobilecoin/blob/master/ledger/db/src/test_utils/mod.rs?

You should be able to initialize a ledger using `initialize_ledger` and add blocks with key images of your choice using `add_block_to_ledger`

Testing with a real ledger is probably going to be better, MockLedger does not necessarily provide the same guarantees and once we add the ledger scanning thread, I think it would be better to test with a real ledger.

Reply:
I think those only work on LedgerDB instances whereas my quotebook is using the ledger trait, so I didn't think I could use this. But after thought, it seems like since this is test code anyway, there isn't a problem specifying that it be a ledgerdb in the test. I can make that change instead of using the mockledger.